### PR TITLE
Fix test on Archlinux

### PR DESCRIPTION
**Pull request by Valantin**
#### Pull Request (PR) description
By default module force db creation by runninf mlocate-updatedb.service but on archlinux the service was updatedb.service

https://wiki.archlinux.org/title/locate

### CODE REVIEW

**Comment by traylenator** [comment]
Any chance to add a regression test to the acceptance tests?

**Comment by Valantin** [comment]
Have you something in mind to check?

**Comment by traylenator** [comment]
> Any chance to add a regression test to the acceptance tests?

Just read the subject, the existing tests was catching it?

File: manifests/config.pp [hunk 1]
```diff
@@ -171,7 +171,7 @@
   if $force_updatedb and $ensure {
     exec { 'force_updatedb':
       command => $_updatedb_command,
-      creates => "/var/lib/${locate}/${locate}.db",
+      unless  => "/usr/bin/test -s /var/lib/${locate}/${locate}.db",
```
Comment (ekohl) [inline manifests/config.pp:174]: Why switch over from `creates`? I'd expect that to work.

Comment (Valantin) [reply manifests/config.pp:174]: on Archlinux /var/lib/plocate/plocate.db exist but empty

so force_updatedb not run and on archlinux acceptance fails

The only way I've found was to check with test if it exist and not empty to run the forced update

**Comment by Valantin** [comment]
> Just read the subject, the existing tests was catching it?

Oh Yes, the resource was idempotent, so if db exist and wasn't empty it skip

we can add test for different locate not only the default defined in hiera to check if all run properly.

I've found mlocate on archlinux use updatedb.timer insteas mlocate-updatedb.timer but I don't know if it was a real problem

**Comment by Valantin** [comment]
> This test should have failed if the database was empty?

Yes, it fails until the database is populated
and by default the database was populated at the end of the configured timer, if you check the result of the modulesync action, it will only fail on archlinux, because update_command was never activated
If I run it manually, the test passes.
It was caused from a existing empty db

File: manifests/config.pp (hunk 1 above)
Comment (ekohl) [reply manifests/config.pp:174]: That's weird. I always interpreted `creates` as an existence check, not a non-empty check. The documentation also only talks about that: https://www.puppet.com/docs/puppet/8/type.html#exec-attribute-creates

The code also suggests that it shouldn't care about (non-)empty: https://github.com/puppetlabs/puppet/blob/fce49baa3767fffb000f1b26d3a7b4edbec58999/lib/puppet/type/exec.rb#L420-L450 (which on UNIX translates to `File.exist?(path)`).

In other words, this shouldn't happen.

I wonder what's in the debug output. We don't run our CI with `BEAKER_PUPPET_DEBUG=1` but that should run puppet with `--debug` providing us with the details that may help.

Comment (Valantin) [reply manifests/config.pp:174]: Try to better explain

Creates check if the file exist if not run the command

On archlinux file exist so the command never run.

Test -s check is file exist and not empty

So it execute if file not exist  like with creates,  and also if it is empty (like in archlinux)

Comment (ekohl) [reply manifests/config.pp:174]: Ah like that. Then it is a behavioral change that should be documented as such.

**Comment by traylenator** [comment]
> > Have you something in mind to check?
> 
> https://github.com/voxpupuli/puppet-mlocate/blob/master/spec%2Facceptance%2Fclass_spec.rb#L38
> 
> This test should have failed if the database was empty?

Found the test failure now  with the other MRs. All makes sense to me.

File: manifests/config.pp (hunk 1 above)
Comment (traylenator) [reply manifests/config.pp:174]: Had a quick play to check if it's possible to generate an empty db file in any way after running updatedb and its impossible since metadata is always saved to the db.

So empty and not existing are equivalent in the context of mlocate - they both indicated that it has  never been run.

**Comment by traylenator** [comment]
I would also say it's a bug in the Arch PKG really. Package should not create an invalid database. If you run `locate` when the database file is empty you get an error.

**Review by zilchms — APPROVED** [review]
